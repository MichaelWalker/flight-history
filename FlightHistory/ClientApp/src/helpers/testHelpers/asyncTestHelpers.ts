import { act } from "@testing-library/react-hooks";

export async function runActionAndWaitForAsyncActionsToComplete(
    action: () => Promise<unknown> | void,
): Promise<void> {
    await act(async () => {
        await action();
    });
}

export async function quickResolve<T>(callback: () => T): Promise<T> {
    return new Promise((resolve) => {
        setTimeout(() => {
            resolve(callback());
        }, 0);
    });
}

export async function quickReject<T>(callback: () => T): Promise<T> {
    return new Promise((resolve, reject) => {
        setTimeout(() => {
            reject(callback());
        }, 0);
    });
}

export class WrappedPromise<T> {
    private readonly promise: Promise<T>;
    private resolved = false;
    private result: T | undefined;

    constructor(promise: Promise<T>) {
        this.promise = promise;

        promise.then((result) => {
            this.resolved = true;
            this.result = result;
        });
    }

    isResolved(): boolean {
        return this.resolved;
    }
}
