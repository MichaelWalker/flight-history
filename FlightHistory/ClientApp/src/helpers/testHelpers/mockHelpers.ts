import * as globals from "../globals";

export function mockUseSampleData(useSampleData: boolean) {
    jest.spyOn(globals, "useSampleData").mockReturnValue(useSampleData);
}

export function mockUseConsoleLogging(useConsoleLogging: boolean) {
    jest.spyOn(globals, "useConsoleLogging").mockReturnValue(useConsoleLogging);
}
