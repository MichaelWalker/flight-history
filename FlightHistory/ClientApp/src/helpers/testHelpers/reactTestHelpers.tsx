import type { ReactElement } from "react";
import React from "react";
import type { RenderResult } from "@testing-library/react";
import { render, screen, waitFor } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";

export function renderWithRouter(component: ReactElement): RenderResult {
    return render(<MemoryRouter>{component}</MemoryRouter>);
}

export async function waitForTableToHaveResults(): Promise<void> {
    await waitFor(() => expect(screen.getAllByRole("row").length).toBeGreaterThan(1));
}
