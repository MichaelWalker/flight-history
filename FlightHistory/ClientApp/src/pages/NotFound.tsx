import React, {FunctionComponent} from "react";

export const NotFoundPage: FunctionComponent = () => {
    return (
        <div>Sorry - that page doesn't exist</div>
    );
};