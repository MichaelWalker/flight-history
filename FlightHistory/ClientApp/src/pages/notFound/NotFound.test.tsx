import "isomorphic-fetch";
import React from "react";
import { renderWithRouter } from "../../helpers/testHelpers/reactTestHelpers";
import { NotFoundPage } from "./NotFound";

describe("NotFound Page", () => {
    it("matches snapshot", async () => {
        const { container } = renderWithRouter(<NotFoundPage />);
        expect(container).toMatchSnapshot();
    });
});
