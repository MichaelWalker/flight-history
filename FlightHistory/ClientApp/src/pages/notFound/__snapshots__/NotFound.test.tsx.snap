// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound Page matches snapshot 1`] = `
<div>
  <div>
    Sorry - that page doesn't exist
  </div>
</div>
`;
